logic/src/commonMain/kotlin/com/wire/kalium/logic/data/backup/BackupRepository.kt

```diff
@@ -86,9 +86,11 @@ internal class BackupDataSource(
                 ),
                 offset = offset,
                 pageSize = PAGE_SIZE.toLong(),
-            ).map {
-                messageMapper.fromEntityToMessage(it)
-            }
+            )
+                .filter { it.expireAfterMs == null }
```

Comment:
let's do it on the query level and not fetch those messages from the DB, it will be much faster than sorting after selecting